### ISSUE TITLE
ValueError: bytes is not a 16-char string

### ISSUE THREAD

**Issue by YasirArfat32** [OpenMined/PySyft#5460]
## Description

The return statement `return deserialize(pb)` in function `get_model(grid_address, worker_id, request_key, model_id)` utilise in the Federated Learning model-centric example `PySyft/examples/federated-learning/model-centric/mcfl_create_plan.ipynb` gives an error `ValueError: bytes is not a 16-char string`

## Software Versions
Pygrid (latest dev branch)
Syft (0.5.0rc1)
Torch (1.8.0)
Torchvision (0.8.2)
Python - (3.7.6)

## How to Reproduce
1. Start Pygrid domain server using command `./run.sh --host localhost  --port 7000 --start_local_db`
2. Execute the NoteBook `PySyft/examples/federated-learning/model-centric/mcfl_create_plan.ipynb`
3. The following error occurs
```
 # Model 
model_params_downloaded = get_model(grid_address, worker_id, request_key, model_id) 
print("Params shapes:", [p.shape for p in model_params_downloaded])                                                                                            
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-22-14a183e0aa75> in <module>
      1 # Model
----> 2 model_params_downloaded = get_model(grid_address, worker_id, request_key, model_id)
      3 print("Params shapes:", [p.shape for p in model_params_downloaded])

<ipython-input-21-43c3f622395f> in get_model(grid_address, worker_id, request_key, model_id)
      6     pb = ListPB()
      7     pb.ParseFromString(req.content)
----> 8     return deserialize(pb)
      9 

~/anaconda3/lib/python3.7/site-packages/syft/core/common/serde/deserialize.py in _deserialize(blob, from_proto, from_bytes)
     87         traceback_and_raise(deserialization_error)
     88 
---> 89     res = _proto2object(proto=blob)
     90     return res

~/anaconda3/lib/python3.7/site-packages/syft/lib/python/list.py in _proto2object(proto)
    156     @staticmethod
    157     def _proto2object(proto: List_PB) -> "List":
--> 158         id_: UID = deserialize(blob=proto.id)
    159         value = []
    160         # list comprehension doesn't work since it results in a

~/anaconda3/lib/python3.7/site-packages/syft/core/common/serde/deserialize.py in _deserialize(blob, from_proto, from_bytes)
     87         traceback_and_raise(deserialization_error)
     88 
---> 89     res = _proto2object(proto=blob)
     90     return res

~/anaconda3/lib/python3.7/site-packages/syft/core/common/uid.py in _proto2object(proto)
    189             if you wish to deserialize an object.
    190         """
--> 191         return UID(value=uuid.UUID(bytes=proto.value))
    192 
    193     @staticmethod

~/anaconda3/lib/python3.7/uuid.py in __init__(self, hex, bytes, bytes_le, fields, int, version, is_safe)
    167         if bytes is not None:
    168             if len(bytes) != 16:
--> 169                 raise ValueError('bytes is not a 16-char string')
    170             assert isinstance(bytes, bytes_), repr(bytes)
    171             int = int_.from_bytes(bytes, byteorder='big')

ValueError: bytes is not a 16-char string
```

## Expected Behavior
Show the expected model information.

## Screenshots

<img width="933" alt="11" src="https://user-images.githubusercontent.com/12051191/114309870-0365e900-9ae9-11eb-95a7-43794d75335d.png">

## System Information
 - OS: [ubuntu]
 - OS Version: [18.04]
 - Language Version: [Python 3.7.6]

**Comment by mnshmnu** [comment]
Having a similar issue at the same point in the notebook

Any solutions yet ? @YasirArfat32

**Comment by YasirArfat32** [comment]
@mnshmnu not yet

**Comment by mnshmnu** [comment]
Have you tried the same with an alternate version of PySyft?

like 0.3.0dev etc..?

@YasirArfat32

**Comment by vvmnnnkv** [comment]
Fixed in https://github.com/OpenMined/PySyft/pull/5520
Note: this error doesn't affect FL model from being hosted or later execution of training plan (in execute plan notenook).
It's just the part of notebook that demonstrates low-level interaction with pygrid and it had outdated code of how model is serialized.

**Comment by vladmihaisima** [comment]
Please reopen as this is not yet fixed in 0.5.0 (latest release) or dev branch. The changes in #5520 still work if applied manually but currently that branch can't be merged to dev due to conflicts.
